bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/discovery/HDPowerViewHubDiscoveryService.java

```diff
@@ -83,17 +90,20 @@ private Runnable createScanner() {
                     NbtAddress address = NbtAddress.getByName(netBiosName);
                     if (address != null) {
                         String host = address.getInetAddress().getHostAddress();
-                        ThingUID thingUID = new ThingUID(THING_TYPE_HUB, host.replace('.', '_'));
+                        String serial = propertyGetter.getSerialNumberApiV1(host);
+                        ThingUID thingUID = new ThingUID(THING_TYPE_HUB, serial);
+                        String label = String.format("@text/%s [\"%s\", \"%s\"]", LABEL_KEY_HUB, "1", host);
                         DiscoveryResult hub = DiscoveryResultBuilder.create(thingUID)
                                 .withProperty(HDPowerViewHubConfiguration.HOST, host)
-                                .withRepresentationProperty(HDPowerViewHubConfiguration.HOST)
-                                .withLabel("PowerView Hub (" + host + ")").build();
-                        logger.debug("NetBios discovered hub on host '{}'", host);
+                                .withProperty(Thing.PROPERTY_SERIAL_NUMBER, serial)
+                                .withRepresentationProperty(Thing.PROPERTY_SERIAL_NUMBER).withLabel(label).build();
+                        logger.debug("NetBios discovered Gen 1 hub '{}' on host '{}'", thingUID, host);
                         thingDiscovered(hub);
                     }
+                } catch (HubException e) {
```

Comment:
Sorry, missed this previously, and very minor, but wouldn't it make sense to simplify and log this case as well (as debug)?
```suggestion
                } catch (HubException | UnknownHostException e) {
```

Reply:
No. I think we should respect the comment below that was written by the original author “nothing to to here”.